mutation/mutation.cc

```diff
@@ -194,6 +196,89 @@ mutation reverse(mutation mut) {
     return *std::move(mut).consume(reverse_rebuilder, consume_in_reverse::yes).result;
 }
 
+namespace {
+class mutation_by_size_splitter {
+    struct partition_state {
+        mutation_rebuilder_v2 builder;
+        size_t empty_partition_size;
+        size_t size = 0;
+        explicit partition_state(schema_ptr schema)
+            : builder(std::move(schema))
+        {
+        }
+    };
+    const schema_ptr _schema;
+    std::vector<mutation>& _target;
+    const size_t _max_size;
+    std::optional<partition_state> _state;
+    template <typename T>
+    stop_iteration consume_fragment(T&& fragment) {
+        const auto fragment_size = fragment.memory_usage(*_schema);
+        if (_state->size && _state->size + _state->empty_partition_size + fragment_size > _max_size) {
+            _target.emplace_back(_state->builder.flush());
+            // We could end up with an empty mutation if we consumed a range_tombstone_change
+            // and the next fragment exceeds the limit. The tombstone range may not have been
+            // closed yet and range_tombstone will not be created.
+            // This should be a rare case though, so just pop such mutation.
```

Comment:
What happens when we flush mutation in between range tombstone changes?

Won't we incorrectly "forget" the range tombstone i.e. it won't end up in any of the mutations we produce?